src/txn/txn_rollback_to_stable.c

```diff
@@ -140,13 +139,13 @@ __rollback_abort_update(WT_SESSION_IMPL *session, WT_ITEM *key, WT_UPDATE *first
               session, key, stable_upd == NULL ? tombstone->start_ts : stable_upd->start_ts));
 
             /*
-             * Clear the history store flag for the first stable update. Otherwise, it will not be
+             * Clear the history store flags for the first stable update. Otherwise, it will not be
              * moved to history store again.
              */
             if (stable_upd != NULL)
-                F_CLR(stable_upd, WT_UPDATE_HS);
+                F_CLR(stable_upd, WT_UPDATE_HS | WT_UPDATE_TO_DELETE_FROM_HS);
             if (tombstone != NULL)
-                F_CLR(tombstone, WT_UPDATE_HS);
+                F_CLR(tombstone, WT_UPDATE_HS | WT_UPDATE_TO_DELETE_FROM_HS);
```

Comment:
I'm a bit uncertain about this. Does it mean that the update no longer needs to be removed from the history store? Is that because the update here that's being rolled back was the reason it needed to be removed in the first place? That connection seems tenuous (as in there isn't a direct link between the cause and effect)

Reply:
bq. Does it mean that the update no longer needs to be removed from the history store?

Yes, that is correct. The update is already removed by rollback to stable. The prepared rollback happens before rts and we mark the update needs to be deleted in the next reconciliation. But before the next reconciliation happens, we run rts which removes the update anyway so the next reconciliation should not remove it again.